portfolio/src/main/webapp/script.js

```diff
@@ -17,9 +17,59 @@
  */
 
 // Spotify API keys to be used for later
-// var client_id = '444dda318d8949cb9282d95a9885a7c6';
-// var client_secret = '9f81ba6cf5748ef9d24a7ddb1e8bc4c';
-// var redirect_uri = 'https://8080-dot-12317386-dot-devshell.appspot.com/misc.html';
+const clientId = "444dda318d8949cb9282d95a9885a7c6";
+let oauthToken = 'BQCebg7W1ZsfuUCdCjdcHz-GDJFMj5GjOhTxvrpNFhu3gvEzYftF3UrSIZ9KHYXVjVdF346qrcaQtd4df7FN0p5sQj2x2p3HpEsCtIDPg4owl4aZx5HfQ4wqWKY3y8l2KakUPS2ZwGKW_PbiPefFMJ3y2FVGfzmz2Kb--w9DxyLBOA';
+
+
+function minutesAndSeconds(milliseconds) {
+  const minutes = Math.floor(milliseconds / 60000);
+  const seconds = ((milliseconds % 60000) / 1000).toFixed(0)
+  return minutes + ':' + (seconds < 10 ? '0' : '') + seconds;
+}
+
+function getCurrentlyPlayingSong() {
+  const response = fetch("https://api.spotify.com/v1/me/player/currently-playing", {
+    method: 'GET',
+    headers: {
+      Authorization: 'Bearer ' + oauth_token
+    }
+  })
+
+  response.then(handleReponse);
+}
+
+setInterval(function() {
+  getCurrentlyPlayingSong();
+}, 500);
+
+function handleReponse(response) {
+
+  // Set variable equal to response json
+  const text = response.json();
+
+  // Pass text to AddSongToPage
+  text.then(addSongToPage)
+}
+
+function addSongToPage(song) {
+  let artists = '';
+  let percentage = (song.progress_ms / song.item.duration_ms) * 100;
```

Comment:
nit: This can be `const` instead of `let`, since percentage is never reassigned.

Actually AFAICT it's never used at all so you can just delete it